File: crates/deepsec-cli/src/commands/status.rs
use crate::config_loader::Context;
use anyhow::Result;
use clap::Args as ClapArgs;
use colored::Colorize;
use deepsec_core::{FileStatus, runs::list_runs, store::load_all_file_records};

#[derive(Debug, ClapArgs)]
pub struct Args {
    #[arg(long)]
    pub project_id: String,
    /// Show last N runs.
    #[arg(long, default_value_t = 5)]
    pub recent: usize,
}

pub fn run(args: Args, ctx: &Context) -> Result<()> {
    let records = load_all_file_records(&ctx.data_root, &args.project_id)?;
    let mut pending = 0usize;
    let mut analyzed = 0usize;
    let mut processing = 0usize;
    let mut error = 0usize;
    let mut total_candidates = 0usize;
    let mut total_findings = 0usize;
    for r in &records {
        match r.status {
            FileStatus::Pending => pending += 1,
            FileStatus::Analyzed => analyzed += 1,
            FileStatus::Processing => processing += 1,
            FileStatus::Error => error += 1,
        }
        total_candidates += r.candidates.len();
        total_findings += r.findings.len();
    }
    println!("{} {}", "project".bold(), args.project_id);
    println!("  records:    {}", records.len());
    println!("  pending:    {pending}");
    println!("  analyzed:   {analyzed}");
    println!("  processing: {processing}");
    println!("  error:      {error}");
    println!("  candidates: {total_candidates}");
    println!("  findings:   {total_findings}");

    let runs = list_runs(&ctx.data_root, &args.project_id)?;
    if !runs.is_empty() {
        println!("\n{}", "recent runs".bold());
        for r in runs.iter().take(args.recent) {
            println!(
                "  {} [{:?}/{:?}] candidates={} findings={}",
                r.run_id,
                r.run_type,
                r.phase,
                r.stats.candidates_found.unwrap_or(0),
                r.stats.findings_count.unwrap_or(0)
            );
        }
    }
    Ok(())
}

File: internal/cli/commands/status.go
package commands

import (
	"fmt"

	"github.com/noeljackson/deepsec/internal/cli"
	"github.com/noeljackson/deepsec/internal/core"
	"github.com/spf13/cobra"
)

// NewStatusCmd prints a per-project summary.
func NewStatusCmd(loader func() (*cli.Context, error)) *cobra.Command {
	var projectID string
	var recent int
	cmd := &cobra.Command{
		Use:   "status",
		Short: "Show pending/analyzed counts and recent runs",
		RunE: func(cmd *cobra.Command, _ []string) error {
			ctx, err := loader()
			if err != nil {
				return err
			}
			records, err := ctx.DataRoot.LoadAllFileRecords(projectID)
			if err != nil {
				return err
			}
			var pending, analyzed, processing, errored, candidates, findings int
			for _, r := range records {
				switch r.Status {
				case core.StatusPending:
					pending++
				case core.StatusAnalyzed:
					analyzed++
				case core.StatusProcessing:
					processing++
				case core.StatusError:
					errored++
				}
				candidates += len(r.Candidates)
				findings += len(r.Findings)
			}
			fmt.Printf("project %s\n", projectID)
			fmt.Printf("  records:    %d\n", len(records))
			fmt.Printf("  pending:    %d\n", pending)
			fmt.Printf("  analyzed:   %d\n", analyzed)
			fmt.Printf("  processing: %d\n", processing)
			fmt.Printf("  error:      %d\n", errored)
			fmt.Printf("  candidates: %d\n", candidates)
			fmt.Printf("  findings:   %d\n", findings)

			runs, err := ctx.DataRoot.ListRuns(projectID)
			if err != nil {
				return err
			}
			if len(runs) > 0 {
				fmt.Println("\nrecent runs")
				lim := recent
				if lim > len(runs) {
					lim = len(runs)
				}
				for _, r := range runs[:lim] {
					var c, f int
					if r.Stats.CandidatesFound != nil {
						c = *r.Stats.CandidatesFound
					}
					if r.Stats.FindingsCount != nil {
						f = *r.Stats.FindingsCount
					}
					fmt.Printf("  %s [%s/%s] candidates=%d findings=%d\n",
						r.RunID, r.Type, r.Phase, c, f)
				}
			}
			return nil
		},
	}
	cmd.Flags().StringVar(&projectID, "project-id", "", "Project id (required)")
	_ = cmd.MarkFlagRequired("project-id")
	cmd.Flags().IntVar(&recent, "recent", 5, "Show last N runs")
	return cmd
}
